File: 010_obj/diferentes_class/Main.java
package diferentes_class;

class FormaBolo {

    String saborMassa;
    String saborRecheio;

    // Construtor
    FormaBolo(String sabor, String recheio) {
        this.saborMassa = sabor;
        this.saborRecheio = recheio;
    }

    // Sobrecarga para recheio padrão
    FormaBolo(String sabor) {
        this(sabor, "sem recheio");
    }

    void mensagemValor() {

        double valor;

        if (saborMassa.equals("limão siciliano") && !saborRecheio.equals("sem recheio")) {
            valor = 63.00;
        } else if (saborMassa.equals("limão siciliano")) {
            valor = 54.00;
        } else if (!saborRecheio.equals("sem recheio")) {
            valor = 41.00;
        } else {
            valor = 31.20;
        }

        System.out.println("""
        ._______________________________________.
        | Bolos dona Donilde
        |.......................................|
        | Sabor: %s - %s
        |.......................................|
        | Total: R$ %.2f
        ._______________________________________.
        """.formatted(saborMassa, saborRecheio, valor));
    }
}

public class Main {
        public static void main(String[] args) {
        FormaBolo bolo = new FormaBolo("limão siciliano");
        bolo.mensagemValor();
    }
}

File: 010_obj/diferentes_class/main.kt
class FormaBolo(
    val saborMassa: String,
    val saborRecheio: String = "sem recheio"
) {

    fun mensagemValor() {

        val valor: Double =
            if (saborMassa == "limão siciliano" && saborRecheio != "sem recheio") {
                63.00
            } else if (saborMassa == "limão siciliano") {
                54.00
            } else if (saborRecheio != "sem recheio") {
                41.00
            } else {
                31.20
            }

        println("""
        ._______________________________________.
        | Bolos dona Donilde
        |.......................................|
        | Sabor: $saborMassa - $saborRecheio
        |.......................................|
        | Total: R$ $valor
        ._______________________________________.
        """.trimIndent())
    }
}

fun main() {
    val bolo = FormaBolo("limão siciliano")
    bolo.mensagemValor()
}
